src/json-observe.ts

```diff
@@ -0,0 +1,262 @@
+/*!
+ * https://github.com/Starcounter-Jack/JSON-Patch
+ * json-patch-duplex.js version: 1.1.2
+ * (c) 2013 Joachim Wester
+ * MIT license
+ */
+declare var Proxy: any;
+declare var Reflect: any;
+declare var module: any;
+
+function _isArray(obj) { return Array.isArray ? Array.isArray(obj) : function (obj) { return obj.push && typeof obj.length === 'number' } };
```

Comment:
Maybe instead of checking `Array.isArray` for every call, we could do it once, and define function or alias it.

Also, If I get it right `_isArray(obj)` 
 - for `Array.isArray == true` returns `Boolean` 
 - for `false` returns `Function`